src/actions/guides/database/migrations.cr

```diff
@@ -0,0 +1,231 @@
+
+    ```crystal
+    def migrate
+      create :users, primary_key_type: :uuid do
+        add email : String
+      end
+    end
+    ```
+
+    > Be sure to update your model with `primary_key_type: :uuid` as well.
+
+    ## Drop table
+
+    To drop a table, use the `drop` method.
+
+    ```crystal
+    def migrate
+      drop :users
+    end
+    ```
+
+    > If your `migrate` method runs `create`, then the `rollback` method should run `drop`.
+
+    ## Alter table
+
+    If your table exists, but you need to make changes to it, use the `alter` method. All of your changes will go in the `alter` block.
+
+    ```crystal
+    def migrate
+      alter :users do
+        remove :old_field
+      end
+    end
+    ```
+
+    ## Add column
+
+    To add a new column, you'll use the `add` method inside of a `[create](#create-table)` block.
+
+    ```crystal
+    create :users do
+      add email : String
+      add birthdate : Time
+      add login_count : Int32
+    end
+    ```
+
+    ### Datatypes
+
+    There are a few different datatypes that you can use to declare the column type.
+
+    * `String` - Maps to postgres `text`.
+    * `Int32` - Maps tp postgres `int`.
+    * `Int64` - Maps to postgres `bigint`.
+    * `Time` - Maps to postgres `timestamptz`.
+    * `Bool` - Maps to postgres `boolean`.
+    * `Float` - Maps to postgres `decimal`. With options for precision and scale.
+    * `UUID` - Maps to postgres `uuid`.
+
+    ### Advanced Options
+
+    The `add` method takes several options for further customization when adding a field.
+
+    * `index` - `false` by default. When `true`, this will create an index on this column.
+    * `using` - `:btree` by default. The index method to use when an index is created.
+    * `unique` - `false` by default. When `true`, postgres will enforce a unique constraint on this field.
+    * `default` - The default value to use for this column.
+
+    ```crystal
+    create :users do
+      add email : String, index: true, unique: true
+      add login_count : Int32, default: 0
+    end
+    ```
+
+    ## Remove column
+
+    The `remove` method must go in the `[alter](#alter-table)` block.
+
+    ```crystal
+    alter :users do
+      remove :middle_name
+      remove :last_login_ip
+    end
+    ```
+
+    ## Add index
+
+    The easiest way is to add the `index: true` option on the `add` method. However, if you're adding indicies after the table is created, you can use the `create_index` method.
+
+    ```crystal
+    def migrate
+      create_index :users, [:status], unique: false
+    end
+    ```
+
+    ## Remove index
+
+    Use the `drop_index` method to remove the index.
+
+    ```crystal
+    def migrate
+      drop_index :users, [:status]
+    end
+    ```
+
+    ## Associations
+
+    If your tables have a one-to-many or a many-to-many relation, you can use these methods to create a foreign key constraint, or remove that constraint from your tables.
```

Comment:
This may be if some help too: https://github.com/luckyframework/lucky/issues/773#issue-442184413

Maybe we can explain this with some examples? I’m fine shipping this as-is but just wanted to reference that in case you’ve got some ideas